main.go

```diff
@@ -116,12 +126,25 @@ func main() {
 		}()
 	}
 
-	server.StatsCollect()
+	stats.Collect()
 	if err := s.Run(); err != nil {
 		log.Fatalf("skydns: %s", err)
 	}
 }
 
+func loadConfig(client *etcd.Client, config *server.Config) error {
```

Comment:
[ Quoting <notifications@github.com> in "Re: [skydns] WIP - Abstract record ..." ]
>>  	if err := s.Run(); err != nil {
>>  		log.Fatalf("skydns: %s", err)
>>  	}
>>  }
>>
>> +func loadConfig(client *etcd.Client, config *server.Config) error {
>
>I tried it but it required gymnastics on initializing the getter (to set priority and ttl once when the getter was created, and then again after the config was loaded).  It felt less coupled this way (I briefly created a second getter but it didn't feel special enough to be independent).

Ok. I will try to refactor this later.

>The etcd logic in main could probably move into backends/etcd as helper code though.

Yes, and while doing that I might extend the `Backend` interface a little (or
not).

/Miek

--
Miek Gieben